spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -538,68 +543,127 @@ protected WriterFactory(Broadcast<Table> tableBroadcast, FileFormat format, long
     @Override
     public DataWriter<InternalRow> createWriter(int partitionId, long taskId, long epochId) {
       Table table = tableBroadcast.value();
-
-      OutputFileFactory fileFactory = OutputFileFactory.builderFor(table, partitionId, taskId).format(format).build();
-      SparkAppenderFactory appenderFactory = SparkAppenderFactory.builderFor(table, writeSchema, dsSchema).build();
-
       PartitionSpec spec = table.spec();
       FileIO io = table.io();
 
+      OutputFileFactory fileFactory = OutputFileFactory.builderFor(table, partitionId, taskId)
+          .format(format)
+          .build();
+      SparkFileWriterFactory writerFactory = SparkFileWriterFactory.builderFor(table)
+          .dataFileFormat(format)
+          .dataSchema(writeSchema)
+          .dataSparkType(dsSchema)
+          .build();
+
       if (spec.isUnpartitioned()) {
-        return new Unpartitioned3Writer(spec, format, appenderFactory, fileFactory, io, targetFileSize);
+        ClusteredDataWriter<InternalRow> dataWriter = new ClusteredDataWriter<>(
+            writerFactory, fileFactory, io,
+            format, targetFileSize);
```

Comment:
Why use a clustered writer? I guess it doesn't matter?

Reply:
It does not matter but should be less overhead.

Reply:
I mean why not just use a `FileWriter` directly?